server/banned.go

```diff
@@ -37,11 +37,12 @@ func (pi *packetInfo) checkBanMessage() error {
 		}
 
 		bannedTrackerData.mutex.Lock()
-		defer bannedTrackerData.mutex.Unlock()
 		bannedTrackerData.bannedBy[td.verificationKey] = nil
+		bannedTrackerData.mutex.Unlock()
 
 		if pi.tracker.banned(bannedTrackerData) {
 			pi.tracker.banIP(bannedTrackerData.conn)
+			pi.tracker.decreasePoolSize(td.pool)
 		}
 
 		return errors.New("player banned")
```

Comment:
I think there is a bug in `messages.go` where it is checking for a more emphatic `"player banned!"` instead of just `"player banned"`. So bans are ignored.

Reply:
Great catch. Fixed!